byzcoin/bcadmin/main.go

```diff
@@ -847,54 +863,84 @@ func link(c *cli.Context) error {
 		if cl == nil {
 			return errors.New("didn't manage to find a node with a valid copy of the given skipchain-id")
 		}
-		ad := &darc.Darc{}
-		adPub := cothority.Suite.Point()
-		// Accept both plain-darcs, as well as "darc:...." darcs
-		adID, err := lib.StringToDarcID(c.String("admindarc"))
-		if err == nil {
-			adPubBuf, err := lib.StringToEd25519Buf(c.String("adminpub"))
+
+		newDarc := &darc.Darc{}
+
+		dstr := c.String("darc")
+		if dstr == "" {
+			log.Info("no darc given, will use an empty default one")
+		} else {
+
+			// Accept both plain-darcs, as well as "darc:...." darcs
+			darcID, err := lib.StringToDarcID(dstr)
 			if err != nil {
-				return err
-			}
-			adPub = cothority.Suite.Point()
-			if err = adPub.UnmarshalBinary(adPubBuf); err != nil {
-				return errors.New("got an invalid admin public key: " + err.Error())
+				return errors.New("failed to parse darc: " + err.Error())
 			}
-			p, err := cl.GetProof(adID)
+
+			p, err := cl.GetProof(darcID)
 			if err != nil {
-				return errors.New("couldn't get proof for admin-darc: " + err.Error())
+				return errors.New("couldn't get proof for darc: " + err.Error())
 			}
-			if err = p.Proof.Verify(id); err != nil {
-				return errors.New("proof for admin is wrong: " + err.Error())
+
+			err = p.Proof.Verify(id)
+			if err != nil {
+				return errors.New("proof for darc is wrong: " + err.Error())
 			}
-			_, adBuf, cid, _, err := p.Proof.KeyValue()
+
+			_, darcBuf, cid, _, err := p.Proof.KeyValue()
 			if err != nil {
 				return errors.New("cannot get value for darc: " + err.Error())
 			}
+
 			if cid != byzcoin.ContractDarcID {
 				return errors.New("please give a darc-instance ID, not: " + cid)
 			}
-			ad, err = darc.NewFromProtobuf(adBuf)
+
+			newDarc, err = darc.NewFromProtobuf(darcBuf)
 			if err != nil {
 				return errors.New("invalid darc stored in byzcoin: " + err.Error())
 			}
 		}
+
+		identity := cothority.Suite.Point()
```

Comment:
So the default identity will always be the null point ? Does it make sense ? And even if it was randomly generated, you wouldn't have the private key associated with it meaning you cannot sign anything. Is there an interest in generating such config ? Same question for the empty darc.

Reply:
Good questions ! Note that this is the original behaviour. 
I found it strange too, but after all I decided to trust the original author...

Reply:
@ineiti Do you remember the reason of this ?

Reply:
yes, the naming is not good (problem #1 in programming). It's not the `default`, but probably `empty` would be better.

It's useful if you want to interact with a byzcoin but don't have a private key associated with a darc. This way you still can `link` with a chain and get the latest blocks or other statistics.

Reply:
I wonder if we should enforce the identity and darc parameters and have a `--passive` or a better name that indicates you only want to interact with the chain. In any case it would nice to have better comments on this as we were two to be confused about it.

Reply:
I updated the comments.